compiler/rustc_typeck/src/check/pat.rs

```diff
@@ -1029,17 +1030,59 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 // last sub-pattern. In the case of `A(x)` the first and last may coincide.
                 // This looks like:
                 //
-                // help: missing parenthesis
+                // help: missing parentheses
                 //   |
                 // L |     let A((x, y)) = A((1, 2));
                 //   |           ^    ^
                 [first, ..] => (first.span.shrink_to_lo(), subpats.last().unwrap().span),
             };
             err.multipart_suggestion(
-                "missing parenthesis",
+                "missing parentheses",
                 vec![(left, "(".to_string()), (right.shrink_to_hi(), ")".to_string())],
                 Applicability::MachineApplicable,
             );
+        } else if fields.len() > subpats.len() {
+            let after_fields_span = if pat_span == DUMMY_SP {
+                pat_span
+            } else {
+                pat_span.with_hi(pat_span.hi() - BytePos(1)).shrink_to_hi()
+            };
+            let all_fields_span = match subpats {
+                [] => after_fields_span,
+                [field] => field.span,
+                [first, .., last] => first.span.to(last.span),
+            };
+
+            // Check if all the fields in the pattern are wildcards.
+            let all_wildcards = subpats.iter().all(|pat| matches!(pat.kind, PatKind::Wild));
+
+            let mut wildcard_sugg = vec!["_"; fields.len() - subpats.len()].join(", ");
+            if !subpats.is_empty() {
+                wildcard_sugg = String::from(", ") + &wildcard_sugg;
+            }
+
+            err.span_suggestion(
+                after_fields_span,
+                "use `_` to explicitly ignore each field",
+                wildcard_sugg,
+                Applicability::MaybeIncorrect,
+            );
+
+            if subpats.is_empty() || all_wildcards {
+                err.span_suggestion(
+                    all_fields_span,
+                    "use `..` to ignore all fields",
+                    String::from(".."),
+                    Applicability::MaybeIncorrect,
+                );
+            } else {
+                err.span_suggestion(
+                    after_fields_span,
+                    "use `..` to ignore the rest of the fields",
+                    String::from(", .."),
+                    Applicability::MaybeIncorrect,
+                );
+            }
```

Comment:
We should create a helper function for `span_suggestions` but with different message, and at the same time open an issue in rustfix that this exists so they can integrate with it.

Reply:
I’m not sure what you’re suggesting. Can you please explain what you mean?

Reply:
I similar helper to `span_suggestion` so we don't need to call it multiple times, as well as being recognized by rustfix, not sure if rustfix is aware that multiple suggestions can be used on the same line right now.

Reply:
See https://github.com/rust-lang/rust/pull/80017#discussion_r546165651.

This doesn’t make sense to do in this PR.

Reply:
There's a precondition to doing this which is reworking the json output schema which I will finally get around to doing this year. It won't be addressable in this PR as it is a ton of unrelated work.